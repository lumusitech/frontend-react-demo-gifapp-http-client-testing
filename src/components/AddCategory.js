import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(inputValue => e.target.value)
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(categories => [inputValue.trim(), ...categories]);
            setInputValue('');
        }
    };

    return (

        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
