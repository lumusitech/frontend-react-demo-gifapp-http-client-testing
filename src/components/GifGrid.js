import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types'

import { GifGridItem } from './GifGridItem'

const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>

            {loading && <p className="card-grid animate__animated animate__flash">Loading...</p>}

            <div className="card-grid animate__animated animate__fadeIn animate__delay-1s">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    )
                    )
                }
            </div>
        </>
    )
}

export default GifGrid

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}