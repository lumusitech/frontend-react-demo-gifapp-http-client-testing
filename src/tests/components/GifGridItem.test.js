import React from 'react';
import { shallow } from 'enzyme';

import '@testing-library/jest-dom'
import { GifGridItem } from '../../components/GifGridItem';

describe('testing on GifGridItem', () => {
    const title = 'Un title';
    const url = 'https://localhost.test.jpg'
    const wrapper = shallow(
        <GifGridItem
            title={title}
            url={url}
        />
    );

    test('should shows the component <GifGridItem /> correctly along time', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('it should has a paragraph with the title', () => {

        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    })

    test('should has an img with url and alt from props', () => {
        const img = wrapper.find('img');

        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    })

    test('should has className animate_fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.props().className;
        expect(className.includes('animate__fadeIn')).toBeTruthy();
    })

})