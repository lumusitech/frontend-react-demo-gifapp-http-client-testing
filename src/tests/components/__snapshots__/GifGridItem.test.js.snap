// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing on GifGridItem should shows the component <GifGridItem /> correctly along time 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un title"
    src="https://localhost.test.jpg"
  />
  <p>
    Un title
  </p>
</div>
`;
